kafka-quickstart/producer/src/main/java/org/acme/kafka/producer/QuotesResource.java

```diff
@@ -0,0 +1,52 @@
+package org.acme.kafka.producer;
+
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.acme.kafka.model.Quote;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.jboss.logging.Logger;
+
+import io.smallrye.mutiny.Multi;
+
+@Path("/quotes")
+public class QuotesResource {
+
+    private static final Logger LOG = Logger.getLogger(QuotesResource.class);
+
+    @Inject
+    @Channel("quote-requests")
+    Emitter<String> quoteRequestEmitter;
+
+    /**
+     * Endpoint to generate a new quote request id and send it to "quote-requests" Kafka topic using the emitter.
+     */
+    @POST
+    @Path("/request")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String createRequest() {
+        UUID uuid = UUID.randomUUID();
+        quoteRequestEmitter.send(uuid.toString());
+        return uuid.toString();
+    }
+
+    @Inject
+    @Channel("quotes")
+    Multi<Quote> quotes;
+
+    /**
+     * Endpoint retrieving the "quotes" Kafka topic and sending the items to a server sent event.
+     */
+    @GET
+    @Produces(MediaType.SERVER_SENT_EVENTS) // denotes that server side events (SSE) will be produced
+    public Multi<Quote> stream() {
+        return quotes.onItem().invoke(quote -> LOG.infov("Received quote {0}", quote));
```

Comment:
```suggestion
        return quotes.log();
```

So, you avoid having to have a logger defined in the class.

Reply:
🤯